pkg/coscheduling/coscheduling.go

```diff
@@ -228,6 +228,21 @@ func (cs *Coscheduling) Permit(ctx context.Context, state *framework.CycleState,
 		waitTime = 0
 	}
 
+	// We will also request to move the sibling pods back to activeQ.
+	if podsToMove := cs.pgMgr.SiblingPods(pod); len(podsToMove) != 0 {
+		c, err := state.Read(framework.PodsToActivateKey)
+		if err == nil {
+			if s, ok := c.(*framework.PodsToActivate); ok {
+				s.Lock()
+				for _, pod := range podsToMove {
+					namespacedName := core.GetNamespacedName(pod)
+					s.Map[namespacedName] = pod
```

Comment:
Should we ignore pods that are already waiting?

Reply:
We can, but IMO it doesn't pay much benefits back. Activate a waiting pod is a no-op, while filtering waiting pods have to iterate the whole waitingPodsMap.

Reply:
In this case, we shouldn't need to exclude the pod itself from SiblingPods either, right?
```go
	for i := range pods {
		if pods[i].UID == pod.UID {
			return append(pods[:i], pods[i+1:]...)
		}
	}
```